File: app-presentation/src/main/java/com/zxjdev/smile/presentation/infrastucture/splash/SplashContract.java
package com.zxjdev.smile.presentation.infrastucture.splash;

import com.zxjdev.smile.presentation.common.base.BasePresenter;
import com.zxjdev.smile.presentation.common.base.BaseView;

public interface SplashContract {

  interface View extends BaseView {

    void navigateToMain();

    void showButtons();
  }

  interface Presenter extends BasePresenter<View> {

    void handleAutoLogin();
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/infrastucture/splash/SplashContract.kt
package com.zxjdev.smile.presentation.infrastucture.splash

import com.zxjdev.smile.presentation.common.base.BasePresenter
import com.zxjdev.smile.presentation.common.base.BaseView

interface SplashContract {

    interface View : BaseView {

        fun navigateToMain()

        fun showButtons()
    }

    interface Presenter : BasePresenter<View> {

        fun handleAutoLogin()
    }
}
